src/app/history/page.tsx

```diff
@@ -0,0 +1,639 @@
+    Record<
+      string,
+      {
+        formattedAmount: string;
+        formattedFee: string;
+      }
+    >
+  >({});
+  const [isFormattingAmounts, setIsFormattingAmounts] = useState(true);
+
+  // Add effect to format amounts when transaction history changes
+  useEffect(() => {
+    if (!transactionHistory?.transactions || isFetchingHistory) return;
+    setIsFormattingAmounts(true);
+
+    const formatTransactions = async () => {
+      const formatted: Record<
+        string,
+        { formattedAmount: string; formattedFee: string }
+      > = {};
+
+      for (const tx of transactionHistory.transactions) {
+        const { parsed } = tx;
+
+        // Format fee
+        const feeResult = await formatAssetAmount({
+          asset: {
+            chainId: parsed.chainId,
+            isToken: false,
+          },
+          amount: parsed.fees.amount,
+          chainData: supportedChains,
+          maximumFractionDigits: 6,
+        });
+
+        // Format amount based on transaction type
+        let amountResult: FormatAssetAmountResult | null = null;
+
+        if (
+          (parsed.mode === "delegate" ||
+            parsed.mode === "undelegate" ||
+            parsed.mode === "claimRewards") &&
+          parsed.validators?.target
+        ) {
+          amountResult = await formatAssetAmount({
+            asset: {
+              chainId: parsed.chainId,
+              isToken: false,
+            },
+            amount: parsed.validators.target.amount,
+            chainData: supportedChains,
+            maximumFractionDigits: 6,
+          });
+        } else if (parsed.mode === "transferToken" && parsed.tokenId) {
+          amountResult = await formatAssetAmount({
+            asset: {
+              chainId: parsed.chainId,
+              isToken: true,
+              assetId: parsed.tokenId,
+            },
+            amount: parsed.recipients?.[0]?.amount || "0",
+            chainData: supportedChains,
+            maximumFractionDigits: 6,
+          });
+        } else if (parsed.recipients?.[0]) {
+          amountResult = await formatAssetAmount({
+            asset: {
+              chainId: parsed.chainId,
+              isToken: false,
+            },
+            amount: parsed.recipients[0].amount,
+            chainData: supportedChains,
+            maximumFractionDigits: 6,
+          });
+        }
+
+        formatted[parsed.id] = {
+          formattedFee: `${feeResult.formatted} ${feeResult.ticker}`,
+          formattedAmount: amountResult
+            ? `${amountResult.formatted} ${amountResult.ticker}`
+            : "",
+        };
+      }
+
+      setFormattedTransactions(formatted);
+      setIsFormattingAmounts(false);
+    };
+
+    formatTransactions();
+  }, [transactionHistory, supportedChains, isFetchingHistory]);
+
+  // Update renderTransaction to use formatted amounts
+  const renderTransaction = (tx: ParsedTransaction) => {
```

Comment:
What you need instead of this, is a new _/src/components/transactions/ParsedTransaction.tsx_ component, that you can use on L604 instead of calling this function.